Source/missiles.cpp

```diff
@@ -2208,105 +2208,54 @@ BOOLEAN __fastcall CheckIfTrig(int x, int y)
 
 void __fastcall AddTown(int mi, int sx, int sy, int dx, int dy, int midir, int mienemy, int id, int dam)
 {
-	int v9;    // ebx
-	int v10;   // esi
-	int v11;   // edi
-	int v12;   // eax
-	int v13;   // ecx
-	char *v14; // eax
-	int v15;   // eax
-	//int v16; // eax
-	int v17;         // ecx
-	int v18;         // eax
-	int v19;         // eax
-	int v20;         // ecx
-	int v21;         // eax
-	int v22;         // ST0C_4
-	int CrawlNum[6]; // [esp+Ch] [ebp-28h]
-	int i;           // [esp+24h] [ebp-10h]
-	char *v25;       // [esp+28h] [ebp-Ch]
-	int v26;         // [esp+2Ch] [ebp-8h]
-	int v27;         // [esp+30h] [ebp-4h]
-	int x;           // [esp+40h] [ebp+Ch]
+	int i, j, k, mx, tx, ty, CrawlNum[6] = { 0, 3, 12, 45, 94, 159 };
 
-	_LOBYTE(v9) = dx;
-	i = mi;
-	v10 = mi;
-	CrawlNum[0] = 0;
-	CrawlNum[1] = 3;
-	CrawlNum[2] = 12;
-	CrawlNum[3] = 45;
-	CrawlNum[4] = 94;
-	CrawlNum[5] = 159;
+	tx = dx;
 	if (currlevel) {
-		_LOBYTE(v11) = dx;
-		missile[v10]._miDelFlag = TRUE;
-		v26 = 0;
-		do {
-			v12 = CrawlNum[v26];
-			v13 = (unsigned char)CrawlTable[v12];
-			v27 = (unsigned char)CrawlTable[v12];
-			if (v13 > 0) {
-				v14 = &CrawlTable[v12 + 2];
-				v25 = v14;
-				while (1) {
-					v9 = dx + (char)*(v14 - 1);
-					v11 = dy + (char)*v14;
-					if (v9 > 0 && v9 < MAXDUNX && v11 > 0 && v11 < MAXDUNY) {
-						v15 = v11 + 112 * v9;
-						if (!(dObject[0][v15] | dPlayer[0][v15] | dMissile[0][v15] | nSolidTable[dPiece[0][v15]] | (unsigned char)nMissileTable[dPiece[0][v15]])) {
-							//_LOBYTE(v16) = CheckIfTrig(v9, v11);
-							if (!CheckIfTrig(v9, v11))
-								break;
+		missile[mi]._miDelFlag = TRUE;
+		for (j = 0; j < 6; j++) {
+			k = CrawlNum[j] + 2;
+			for (i = (unsigned char)CrawlTable[CrawlNum[j]]; i > 0; i--) {

```

Comment:
```suggestion
			for (i = 0; i < (unsigned char)CrawlTable[CrawlNum[j]]; i++) {
```

Reply:
@AJenbo, the suggestion you provided changes the loop from `i--` to `i++`. Which is the right one? (I'm assuming `i++`, as otherwise the loop is a nop?)

Reply:
Actually, looking closer at it. It should be `i--`, not `i++`. Since `i` starts at a value above 0.

**Edit:** never mind. My brain hasen't woken up yet. Both loops are correct. @qndel loop starts at `i = Crawl..` and @AJenbo's at `i = 0`. If the suggestion by @AJenbo is bin exact, lets go with that one. Otherwise, the code looks fine as is.